### ISSUE TITLE
Context plugin doesn't detect SSH sessions

### ISSUE THREAD

**Issue by BrainStone** [romkatv/powerlevel10k#1723]
When using the context plugin it does not seem to properly detect SSH connections.

For example with having this in the `.p10k.zsh` file the non remote templates are used for both local shells and remote shells:

```zsh
typeset -g POWERLEVEL9K_CONTEXT_ROOT_TEMPLATE='%n@%m (local root)'
typeset -g POWERLEVEL9K_CONTEXT_{REMOTE,REMOTE_SUDO}_TEMPLATE='%n@%m (remote)'
typeset -g POWERLEVEL9K_CONTEXT_TEMPLATE='%n@%m (local)'
```

I do have the theme installed through Oh My ZSH in a custom directory. The plugin is enabled for the right side.  
My full configuration can be found here: https://github.com/BrainStone/zsh-customization  
And my full `.p10k.zsh` config here (pinned to the current state): https://github.com/BrainStone/zsh-customization/blob/874a89703a7134749555867d59eb11542d2c5b3f/p10k_config.zsh#L909-L914

The variable `SSH_TTY` is set during my SSH session, and not during the local session.

**Comment by romkatv** [comment]
What's the output of the following command when connected over SSH?

```zsh
print -P '%#'
```

**Comment by BrainStone** [comment]
> What's the output of the following command when connected over SSH?
> 
> ```shell
> print -P '%#'
> ```

The output is `#` for a session where I am connecting directly as root.  
Interestingly enough when connect to an user account instead of directly as root, the context is correct (and for that case the output is `%`)

**Comment by romkatv** [comment]
See comments in your config. Here they are:

```zsh
# Context format when running with privileges: bold user@hostname.
typeset -g POWERLEVEL9K_CONTEXT_ROOT_TEMPLATE='%B%n@%m'
# Context format when in SSH without privileges: user@hostname.
typeset -g POWERLEVEL9K_CONTEXT_{REMOTE,REMOTE_SUDO}_TEMPLATE='%n@%m'
# Default context format (no privileges, no SSH): user@hostname.
typeset -g POWERLEVEL9K_CONTEXT_TEMPLATE='%n@%m'
```

`root` user is privileged.

If you want to customize based on SSH alone, use this construct:

```zsh
if (( P9K_SSH )); then
  typeset -g ...
else
  typeset -g ...
fi
```

**Comment by BrainStone** [comment]
Thank you very much! Though I was under the impression, that it would work at least for when sudoing in SSH sessions.

**Comment by romkatv** [comment]
It does work when sudoing in SSH sessions. SSH gets detected correctly, and so does the privileged bit. Styling is applied exactly as the comments say.

**Comment by BrainStone** [comment]
> It does work when sudoing in SSH sessions. SSH gets detected correctly, and so does the privileged bit. Styling is applied exactly as the comments say.

When I test that it still selects the `POWERLEVEL9K_CONTEXT_ROOT_TEMPLATE` when being logged in through SSH and then having run `sudo -i`:

![grafik](https://user-images.githubusercontent.com/4106544/149673301-f6d83f78-fc28-4628-a538-fda1b76e15dc.png)

Using this config to be able to differentiate between all 3 cases

```zsh
typeset -g POWERLEVEL9K_CONTEXT_ROOT_TEMPLATE='%n@%m (local root)'
typeset -g POWERLEVEL9K_CONTEXT_{REMOTE,REMOTE_SUDO}_TEMPLATE='%n@%m (remote)'
typeset -g POWERLEVEL9K_CONTEXT_TEMPLATE='%n@%m (local)'
```

**Comment by romkatv** [comment]
Your screenshot looks as expected to me. Do you believe otherwise?

**Comment by BrainStone** [comment]
> Your screenshot looks as expected to me. Do you believe otherwise?

Well yes. I sudoed in a SSH session, so I expected the `POWERLEVEL9K_CONTEXT_REMOTE_SUDO_TEMPLATE` to be applied. Or is it meant for when I ssh from a root user?

**Comment by romkatv** [comment]
> I sudoed in a SSH session, so I expected the `POWERLEVEL9K_CONTEXT_REMOTE_SUDO_TEMPLATE` to be applied.

These parameters are applied exactly as the comments say. The comment above `POWERLEVEL9K_CONTEXT_REMOTE_SUDO_TEMPLATE` says: "Context format when in SSH without privileges." You are running as root, hence with privileges, so this parameter does not apply. The comment above `POWERLEVEL9K_CONTEXT_ROOT_TEMPLATE` says: "Context format when running with privileges." You are running as root, hence with privileges, so this parameter does apply.

**Comment by BrainStone** [comment]
Ok. Makes sense then why these two are grouped. But what exactly is the difference between `POWERLEVEL9K_CONTEXT_REMOTE_SUDO_TEMPLATE` and `POWERLEVEL9K_CONTEXT_REMOTE_TEMPLATE`?

According to the docs from powerlevel9k (I know this is the successor, but from what I've read so far in the docs it is supposed to keep the config options to work as closely to the original as possible) the `REMOTE_SUDO` format should appear when SSH'ed into a server and having elevated rights: https://github.com/Powerlevel9k/powerlevel9k#context

**Comment by romkatv** [comment]
> Ok. Makes sense then why these two are grouped. But what exactly is the difference between `POWERLEVEL9K_CONTEXT_REMOTE_SUDO_TEMPLATE` and `POWERLEVEL9K_CONTEXT_REMOTE_TEMPLATE`?

There is no practical difference. Please trust me that these parameters work as the comments say.

> According to the docs from powerlevel9k (I know this is the successor, but from what I've read so far in the docs it is supposed to keep the config options to work as closely to the original as possible) the `REMOTE_SUDO` format should appear when SSH'ed into a server and having elevated rights: https://github.com/Powerlevel9k/powerlevel9k#context

In this regard p10k works just like p9k. This is the reason these parameters have such awful names. Unlike p9k, p10k has correct documentation for these parameters.

**Comment by BrainStone** [comment]
Ok. Yeah it seems like the names are pretty awful, considering the behavior I experienced being in stark contrast to the names of the variables (though the comments being right).

I apprechiate your time for both looking into this issue, pointing out the incorrect docs in powerlevel9k, providing me a snippet that does what I tried to achieve and generally making this amazing theme. I hope one day I give back, but until then, thank you very much!
